<!doctype html>
<html lang="en">
<head>
  <!-- Required meta tags -->
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="author" content="Ryohei Suzuki">
  
  <title>Treefit - Quantitative analysis software for trajectory inference</title>
    <!-- Measuring the tree-likeness of single-cell RNA-seq data -->
  
  <!-- Bootstrap CSS -->
  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
  
  <script src="https://code.jquery.com/jquery-3.4.1.slim.min.js" integrity="sha384-J6qa4849blE2+poT4WnyKhv5vZF5SrPo0iEjwBvKU7imGFAV0wwj1yYfoRSJoZ+n" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
  <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js" integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6" crossorigin="anonymous"></script>
  
  <!-- custom css -->
  <link href="css/main.css" rel="stylesheet">
</head>
<body>
  <!-- ナビゲーションバー -->
  <header>
    <div class="navbar navbar-dark bg-dark shadow-sm">
      <div class="container d-flex justify-content-between">
        <a href="#" class="navbar-brand d-flex align-items-center">
          <strong>treefit</strong>
        </a>
        <span>
          <a href="#" class="link-github">
            <img src="img/GitHub-Mark-Light-64px.png" alt="github repository">
          </a>
        </span>
      </div>
    </div>
  </header>
  <main role="main">
    <!-- タイトルロゴ（仮） -->
    <p class="text-center"><img src="img/treefitlogo.png" width=1000></p>
    <!-- タイトル部分 -->
    <!--
    <section class="jumbotron text-center">
      <div class="container">
        <h1>Treefit</h1>
        <h3>Quantitative analysis software for trajectory inference</h3>
      </div>
    </section>
    -->
    <!-- Key features 部分 -->
    <div class="text-center">
      <div class="col-md-8 offset-md-2">
        <h2>Key features of Treefit</h2>
      <p>
      <b>Treefit</b> is a data analysis toolkit that helps you 1) assess the goodness-of-fit of tree models to your single-cell gene expression data and 2) decide which tree best fits your data. <b>Treefit</b> is implemented in both R and Python languages and can be used in conjunction with existing popular software packages, such as Monocle, Seurat, and Dynverse.
      </p>
        </div>
    </div>
    <!-- アブストラクト部分 -->
    <section class="row" id="section-abstract">
      <div class="col-md-5 offset-md-1 text-center">
        <h3>1) Verifying the goodness-of-fit between data and tree models</h3>
        <p>
          Trajectory inference is one of the key computational challenges in single-cell transcriptomics. In the last few years, many software packages have been developed and widely used to estimate and visualize the underlying tree structure (<i>i.e.</i>, cellular hierarchy) from single-cell RNA-seq data. 
          However, trajectory inference often suffers from the uncertainty due to the heterogeneity of individual cells and the high levels of technical noise in single-cell experiments. Therefore, in order to facilitate scientific discoveries, it is crucial to establish a theoretically sound quantitative method that is underpinned by a mathematical/statistical evidence base. 
          <b>Treefit</b> is a library to examine the reliability of inferred trajectories based on the robustness of the tree structures.
        </p>
          <div class="abstract-figure-container">
          <img src="img/abstract-left.png" class="abstract-figure">
        </div>
      </div>
      <div class="col-md-5 text-center">
        <h3> 2) Estimating the number of principal cell linages in your dataset</h3>
        <p>
          Single-cell technologies can help us discover a novel type of cells and revolutionalize our understanding of the process of cell differentiation; however, it is not easy to detect distinct types of cells or to assess how different their gene expression profiles are. Although visualization is an essential part of any exploratory data analysis, the output images do not always indicate a scientific truth. With help of powerful techniques in different disciplines of mathematics, <b>Treefit</b> gives insight into the "shape" of your data (<i>e.g.</i> linear maturation, bifurcation, or more complex divergence) and helps accelerate scientific discoveries in a mathematically and statistically valid way.  
        </p>
        <div class="abstract-figure-container">
          <img src="img/abstract-right.png" class="abstract-figure">
        </div>
      </div>
    </section>
    <!-- 理論紹介部分 -->
    <section id="section-theory" class='bg-light'>
      <div class="text-center">
         <div class="col-md-8 offset-md-2">
        <h2>Theoretical ideas</h2>
        <p>One of the fundamental assumptions in biology is that cell differentiation can be modeled by a tree (<i>i.e.</i>, connected graph without a cycle).  <b>Treefit</b> verifies how well your data can be explained by a tree the fitting of the data to a tree topology exploiting the uncertainty of data.</p>
          </div>
      </div>
      <div class="row">
        <div class="col-md-8 offset-md-2">
          <div class="step">
            <h3>1. Applying perturbation</h3>
            <p><b>Treefit</b> repeatedly perturbs the given data to generate many possible realizations of gene expression matrix.</p>
            <p>It uses the statistical uncertainty of the raw counts as well as an interpolating techniques for the perturbation.</p>
            <p class="text-center"><img src="img/theory-perturbation.png" width=500></p>
          </div>
          <div class="step">
            <h3>2. Comparing the global structures of trees</h3>
            <p><b>Treefit</b> calculates the fitted tree (currently supporting MST) for each realized expression matrix, then compares their <i>global structures</i> using geometrical methods.
            The structure is stable when a data has an intrinsic tree-like topology, while it becomes unstable for not tree-like data (e.g., cycle, diffusive).</p>
            <p class="text-center"><img src="img/theory-msts.png" width=550></p>
            <p><i>comparison of stable/unstable MSTs generated from tree-like/not tree-like data (only longest paths are shown):</i></p>
            <p class="text-center"><img src="img/theory-comparison.png" width=700></p>
          </div>
          <div class="step">
            <h3>3. Grassmann manifolds of Laplacian eigenseries</h3>
            <p>As the measure of the similarity of global structures between trees, <i>treefit</i> calculates the eigenseries of the graph Laplacian for each tree, and then computes the distance between the spaces spanned by the leading eigenvectors, which represents the low-frequency components of the graph topology.</p>
            <p>Several distance measures with different definitions provides diverse information about the topology of the data.</p>
            <p class="text-center"><img src="img/theory-grassmann.png" width=800></p>
          </div>
        </div>
      </div>
    </section>
    <!-- チュートリアル部分 -->
    <section id="section-getting-started">
      <div class="text-center">
        <h2>Getting started</h2>
      </div>
      <div class="row">
        <div class="col-md-8 offset-md-2">
          <h3>install</h3>
          <div class="step">
            <table class="code-sample">
              <tr>
                <td><span class="lang-icon lang-icon-red">R</span></td>
                <td>
                  <pre>install.packages("treefit")
library(treefit)</pre>
                </td>
              </tr>
            </table>
          </div>
          <h3>prepare count data as a <a href="https://satijalab.org/seurat/" target="_blank">Seurat</a> object</h3>
          <div class="step">
            <table class="code-sample">
              <tr>
                <td><span class="lang-icon lang-icon-red">R</span></td>
                <td>
                  <pre>count &lt;- Read10X(data.dir = "/path/to/data/dir")
data &lt;- CreateSeuratObject(counts = pbmc.data)</pre>
                </td>
              </tr>
            </table>
          </div>
          <h3>run <i>treefit</i></h3>
          <div class="step">
            <table class="code-sample">
              <tr>
                <td><span class="lang-icon lang-icon-red">R</span></td>
                <td>
                  <pre>treefit_result &lt;- treefit(data)</pre>
                </td>
              </tr>
              <tr>
            </table>
            <p><code>treefit</code> performs the <a href="https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html">standard pre-processing</a> workflow as the default behavior during the computation.</p>
            <p>User can also pass a custom protocol as an optional argument. See <a href="#">tutorial</a> for the details.</p>
          </div>
          <h3>see the result</h3>
          <div class="step">
            <table class="code-sample">
              <tr>
                <td><span class="lang-icon lang-icon-red">R</span></td>
                <td>
                  <pre>
ggplot(data = treefit_result) + geom_line(mapping = aes(x=k, y=max))
                              + geom_line(mapping = aes(x=k, y=mean))</pre>
                </td>
              </tr>
              <tr>
            </table>
            <p><code>treefit</code> returns a dataframe containing a series of values called <i>Grassmann distances</i>, from which we can learn how the input data fits a tree-like topology.</p>
            <ul>
              <li><code>max</code> indicates the <i>tree-likeness</i> (confidence of the fitting to a tree topology) of the data.</li>
              <li><code>mean</code> tells us the intrinsic complexity of the data by its function shape.</li>
            </ul>
            <p>
              from the result below, we can learn that the given data is highly tree-like, and has a bifurcation-like topology.
            </p>
            <p class='text-center'>
              <img src="img/getstart-result-sample.png" width=400>
            </p>
            <p>See <a href="#">tutorial</a> for the detailed usage and the interpretation of the result.</p>
          </div>
        </div>
      </div>
    </section>
  </main>
  <footer class="text-muted">
    <div class="container">
      <p class="float-right">
        <a href="#">Back to top</a>
      </p>
      <p>Last update: Feb. 7, 2020</p>
      <p>&copy; Hayamizu Lab, The Institute of Statistical Mathematics, Japan</p>
    </div>
  </footer>
</body>
</html>
