---
title: Treefit - The first software for quantitative trajectory inference
description: Treefit - The first software for quantitative trajectory inference
layout: base
---

<section id="section-title">
  <div class="title-left">
    <div class="title-right">
      <div class="text-center title-content">
        <div class="container">
          <h1>Treefit</h1>
          <h2>The first software for <em>quantitative</em> trajectory inference</h2>
        </div>
      </div>
    </div>
  </div>
</section>

<section id="section-abstract">
  <div class="container">
    <div class="row">
      <div class="col-md-10 offset-md-1 text-center">
        <p>
          <b>Treefit</b> is a novel data analysis toolkit that helps you perform two types of quantitative analysis: 1) measuring the goodness-of-fit between your single-cell RNA-seq data and esimated tree trajectories; and 2) discovering novel cell types or detecting contaminating cell types by predicting the number of principal paths in the best-fit tree trajetory. <b>Treefit</b> can be used to analyze either row counts or normalized expresion data. <b>Treefit</b> is implemented in both R and Python languages and can be used in conjunction with other  popular software packages, such as Seurat and dynverse. 
        </p>
      </div>
    </div>
  </div>

  <div class="container">
    <div class="row">
      <div class="col-md-5 offset-md-1">
        <h3>1) Measuring the goodness-of-fit between data and data-derived tree trajectories</h3>
        <p>
          Trajectory inference is one of the key computational challenges in single-cell transcriptomics. In the last few years, many software packages have been developed and widely used to extract an underlying "tree" structure from single-cell RNA-seq data.
        </p>
        <p>
          However, trajectory inference often suffers from the uncertainty due to the heterogeneity of individual cells and the high levels of technical noise in single-cell experiments. Therefore, in order to facilitate scientific discoveries, it is crucial to establish a theoretically sound quantitative method that is underpinned by a mathematical/statistical evidence base.
        </p>
        <p>
          <b>Treefit</b> is a library to examine the reliability of inferred trajectories based on the robustness of the tree structures.
        </p>
        <div class="abstract-figure-container">
          <img src="/assets/images/abstract-left.png" class="abstract-figure">
        </div>
      </div>
      <div class="col-md-5">
        <h3>2) Estimating the number of "principal paths" in the best-fit tree trajectory</h3>
        <p>
          Single-cell technologies are expected to help us discover a novel type of cells and revolutionize our understanding of the process of cell differentiation; however, it is not always easy to distinguish distinct types of cells or to quantify how different their gene expression profiles are.
        </p>
        <p>
           Many past studies involved exploratory data analysis relying on visualization. However, using different visualization techniques can produce completely different pictures and the interpretation can be subjective. Therefore, there has been an urgent need for a more objective, reliable methodology in order to directly reach a scientific truth.
        </p>
        <p>
          With help of powerful techniques in various disciplines of mathematics, <b>Treefit</b> gives insight into the "shape" of your data (<i>e.g.</i> linear maturation, bifurcation, or more complex divergence) and helps accelerate scientific discoveries in a mathematically and statistically valid way.
        </p>
        <div class="abstract-figure-container">
          <img src="/assets/images/abstract-right.png" class="abstract-figure">
        </div>
      </div>
    </div>
  </div>
</section>

<section id="section-theory" class="bg-light">
  <div class="container">
    <h2>Theoretical ideas</h2>
    <p>One of the fundamental beliefs in cell biology is that the process of cell differentiation can be modeled by a "tree" (<i>i.e.</i>, connected graph without a cycle). However, it depends on the situation whether or not this assumption is valid. Indeed, some data are too noisy to be exactly depicted by a tree and some cells are known to undergo dedifferentiation that is not a tree-like but cycle-like process. <b>Treefit</b> verifies how well your data can be explained by a tree model and thus helps you perform quantitative trajectory inference.</p>
    <div class="step">
      <h3>1. Adding small noises to original data</h3>
      <p>In mathematics, perturbation refers to the operation of adding a slight shift to a number. In our context, it means adding small noises to the values of original gene expression data. If you input a gene expression data matrix, then <b>Treefit</b> repeatedly perturbs the data so that you can obtain many possible datasets without actually conducting the same single-cell experiments over and over again.  
      <p><b>Treefit</b> currently supports two types of perturbation, one of which uses the statistical uncertainty of the raw counts while the other employs an interpolating techniques that can also be applied to processed data.</p>
      <p class="text-center"><img src="/assets/images/theory-perturbation.png"></p>
    </div>
    <div class="step">
      <h3>2. Measuring the robustness of inferred tree trajectories</h3>
      <p>Once creating many datasets by simulating different kinds of noise, we can build many possible trees by applying any tree trajectory inference method to each gene expression data matrix. <b>Treefit</b> calculates a tree for each realized expression matrix, then compares their <i>global structures</i> using geometrical methods. The current version of <b>Treefit</b> computes a minimum spanning tree (MST) because many existing trajectory inference software packages uses MSTs.</p>
      <p class="text-center"><img src="/assets/images/theory-msts.png"></p>
      <p><b>Treefit</b> analyzes the goodness-of-fit between data and data-derived tree models by measuring the structural similarity between many inferred tree trajectories. In principle, the tree structure is stable when data have an intrinsic tree-like topology, while it becomes unstable for non-tree-like data (<i>e.g.</i>, cycle, diffusive).</p>
      <p class="text-center"><img src="/assets/images/theory-comparison.png"></p>
      <p><i>The analysis of two artificial datasets shows that the structure of MSTs is robust to noise when the shape of data is tree-like but is not in the cycle-like case (note that longest paths are highlighted for illustrative purposes here).</i></p>
    </div>
    <div class="step">
      <h3>3. Grassmann manifolds of Laplacian eigenseries</h3>
      <p>As the measure of the similarity of global structures between trees, <b>Treefit</b> calculates the eigenseries of the graph Laplacian for each tree, and then computes the distance between the spaces spanned by the leading eigenvectors, which represents the low-frequency components of the graph topology.</p>
      <p><b>Treefit</b> supports several distance measures with different definitions in order to help extract diverse information about the topology of the data.</p>
      <p class="text-center"><img src="/assets/images/theory-grassmann.png"></p>
    </div>
  </div>
</section>

<section id="section-getting-started">
  <div class="container">
    <div class="text-center">
      <h2>Getting started</h2>
      <p>There are <b>Treefit</b> for R and <b>Treefit</b> for Python.</p>
    </div>
    <div class="row">
      <div class="col-md-6">
        <p class="text-center"><a href="/treefit-r/"><b>Treefit</b> for R</a></p>
      </div>
      <div class="col-md-6">
        <p class="text-center"><a href="python/"><b>Treefit</b> for Python</a></p>
      </div>
    </div>
  </div>
</section>
